contributors/design-proposals/add-metadata-indicating-to-replace-union-in-patch.md

```diff
@@ -0,0 +1,280 @@
+Add Tag Indicating To Replace *Union* Fields in Patch (a.k.a Support Replacing Maps in Strategic Merge Patch)
+=============
+
+This approach could be used to fix the issues with `kubectl apply` simply by having apply always replace union fields instead of patching them.  E.g. instead of setting `spec.strategy.type` without touching the rest of `spec.strategy`, patches should always replace the entire `spec.strategy`.
+
+## Proposed Changes
+
+### APIs
+
+Add a go struct tag/openAPI value telling clients to replace the entire struct when doing a *PATCH*. For the *inline* fields, we have 2 options:
+
+1) Move the tag to the parent struct's. E.g. add a tag to `Volume` instead of `VolumeSource`. This will limit the flexibility to add more fields in the future.
```

Comment:
@thockin I think that is the proposal here.  The compromise would be that each Volume would need to be entirely managed by apply + configuration file or not managed by it at all.  e.g. A Volume could not have some fields managed by apply and come fields managed by another writer (e.g. some controller).